dcos/dcos-ui
# 🤫 feat: parameterized root marathon

## Outstanding issues:

- [ ] Images are broken (i.e. Mesosphere Logo)
- [ ] Update service tries calling the relative path -> Config.rootUrl most probably should be set to "/" or we rather should disable the update plugin
- [ ] Disable Catalog
- [ ] Some kind of indication that UI is running on MOM
---

Introducing 2 global variables to parameterize root Marathon.
With the change it will be possible directing the UI to a MoM (Marathon on Marathon) so that operators could operate MoMs easily.

## Testing

1. Launch `marathon` from the Catalog usually it goes under name `marathon-user`
2. Set `window.MARATHON_NAME = "marathon-user";`
3. Set `window.MARATHON_API_PREFIX = "/service/marathon-user/v2";`
4. Check the UI, it should be _empty_ unless something is running on the MoM already
5. Launch an app and a pod
6. See screenshots

## Trade-offs

Oh well, global variables

## Dependencies

None

## Screenshots
Left screenshot: docs-ui pointing to the root Marathon
Right screenshot: dcos-ui pointing to mom
![Screenshot 2019-04-17 at 10 43 12](https://user-images.githubusercontent.com/186223/56284385-9b0be900-6114-11e9-82ab-cc5c4a2ac0a4.png)
